index.ts

```diff
@@ -295,9 +296,25 @@ export default class Analytics {
       return this.nullFlushResponse();
     }
 
-    const items = this.queue.splice(0, this.flushAt);
-    const callbacks = items.map((item) => item.callback);
-    const messages = items.map((item) => {
+    let flushSize = 0;
+    let spliceIndex = 0;
+
+    // guard against requests larger than 4mb
+    this.queue.slice(0, this.flushAt).every((item) => {
```

Comment:
could also work as a for loop

Reply:
Write out the for-loop here -- better to use every() with pure functions.